src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -218,7 +219,28 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                             targetsList = await GetTargetsListAsync(assetsFilePath, token);
                         }
 
-                        transitiveOrigins = calculatedTransitivePackages.Any() ? ComputeTransitivePackageOrigins(calculatedInstalledPackages, targetsList, token) : new Dictionary<string, TransitiveEntry>();
+                        // If the project has project references, we need to compute transitive origins for their packages
+                        List<LockFileTargetLibrary> projectReferences = targetsList.SelectMany(t => t.Libraries).Where(l => l.Type == LibraryType.Project).ToList();
```

Comment:
This is just a comment, but I'm wondering how many of our past failures in this logic stem from the fact that dependencies from different frameworks are combined together. 

It can be a separate work, but I'm wondering what would happen if project references are different.

Reply:
It wouldn't surprise me if conditional project references are a common scenario for MAUI projects. Or at least, that's where I feel like it's most likely to happen.

Reply:
Conditional package references are very common in MAUI.

Reply:
What happens when they package references are conditional? I've been trying to test with them and looks like they are not being added to the `Libraries` section of the assets file, is that something expected? I do see the project reference in the Solution Explorer

Reply:
The assets file there doesn't load the libraries section: https://github.com/NuGet/NuGet.Client/blob/0c714c10851db160a4ca5b80e718612bf669daf3/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs#L414.

Reply:
i'm guessing you mgiht've referred to https://github.com/NuGet/NuGet.Client/blob/0c714c10851db160a4ca5b80e718612bf669daf3/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileTarget.cs#L19. 


The solution explorer reads the assets file, so it probably should be there.

Reply:
Sorry I meant to say Target section, are they expected to be there?

Reply:
Yes.

Reply:
Tested with conditional ProjectReferences, we will only show transitive packages from a project reference if they are the same target framework as the one the PM UI is using (I believe is the first one in the list, not sure if we order them).

What do you suggest it could be a next step? I came with another idea other than supporting tfm's in the PM UI

Reply:
A referenced project that supports TFMs that the current project does not, will not cause transitive packages through the unsupported TFM.  So, I don't understand the scenario. For example, say ProjectA supports net48 and references System.Memory, and also targets net8.0 without any package references. ProjectB has a ProjectReference to ProjectA, but ProjectB only targets net8.0.  ProjectB will not have a transitive reference to System.Memory and therefore should not be shown in ProjectB's PM UI (it also won't be in the assets file).

If ProjectB multi-targets the same TFM as ProjectA, then System.Memory will be a transitive package, and should be shown in PM UI (it will be in the assets file, but only for the net48 TFM)

Reply:
I don't expect a change in behavior for installed packages since this PR only changes how we calculate transitive origins. These ProjectReferences are added to a copy and used only for origins calculations.

Reply:
We should probably check how thorough our automated/manual tests are on this topic. 

It doesn't necessarily have to be blocking this PR, but as we're getting transitives in the solution level PM UI soon, ensuring proper coverage is really important.